File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.rb
# Definition for a binary tree node.
# class TreeNode
#   attr_accessor :val, :left, :right
#   def initialize(val = 0, left = nil, right = nil)
#     @val = val
#     @left = left
#     @right = right
#   end
# end
# @param {TreeNode} root
# @return {Integer[]}
def postorder_traversal(root)
  ans, popped = [], nil
  stack = root ? [root] : []
  while !stack.empty?
    popped = stack.pop
    ans.unshift(popped.val)
    stack << popped.left if popped.left
    stack << popped.right if popped.right
  end
  ans
end

File: 0145-binary-tree-postorder-traversal/0145-binary-tree-postorder-traversal.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *   Val int
 *   Left *TreeNode
 *   Right *TreeNode
 * }
 */
func postorderTraversal(root *TreeNode) []int {
  ans, stack := []int{}, []*TreeNode{}
  if root != nil {
    stack = append(stack, root)
  }
  var popped *TreeNode
  for len(stack) > 0 {
    popped = stack[len(stack) - 1]
    stack = stack[:len(stack) - 1]
    ans = append([]int{(*popped).Val}, ans...)
    if (*popped).Left != nil {
      stack = append(stack, (*popped).Left)
    }
    if (*popped).Right != nil {
      stack = append(stack, (*popped).Right)
    }
  }
  return ans
}